contrib/babelfishpg_tsql/runtime/functions.c

```diff
@@ -4455,32 +4457,79 @@ bbf_pivot(PG_FUNCTION_ARGS)
 				(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
 				 errmsg("materialize mode required, but it is not allowed in this context")));
 	
-	if (fcinfo->context == NULL || !IsA(fcinfo->context, List) || list_length((List *) fcinfo->context) != 3)
-		ereport(ERROR,
-			(errcode(ERRCODE_CHECK_VIOLATION),
-				errmsg("Babelfish PIVOT is not properly initialized.")));
+	bbf_pivot_src_sql = NULL;
+	bbf_pivot_cat_sql = NULL;
+	query_string = NULL;
+	funcName = NULL;
+
+	/*
+	 * if view_uuid is not defined (view_uuid is empty string), then we will get the pivot metadata from
+	 * FunctionCallInfo(fcinfo).
+	 */
+	if (strlen(view_uuid) == 0)
+	{	
+		if (fcinfo->context == NULL || !IsA(fcinfo->context, List) || list_length((List *) fcinfo->context) != 3)
```

Comment:
nit: I am not a huge fan of hardcoding values, in this case `3`. You can define a constant with descriptive name and use that instead.

Reply:
defined a constant called `PIVOT_METADATA_COUNT`